### ISSUE TITLE
[slim] update README.md with more detailed and easy understandable examples

### ISSUE THREAD

**Issue by abes975** [tensorflow/models#411]
It's not clear from the documentation how to handle batch size while  using slim.learning.train
It would be very helpful to provide an example of training with data partitioned in batches. (for example to face memory allocation problems)
Does learning.train handle splitting  data in batches  or this has to be done by manually with an explicit loop? 
There is  very quick snippet that uses
`images, labels = tf.train.batch([image, label])`
bur for newbies is not clear at all how to create `image` and `label` tensor (seems to be contain only a single element from the name), because the reading data function is omitted.
Is it possible to provide a very simple complete example starting from 2 numpy arrays that containing all training_data and training_labels ?

**Comment by evolu8** [comment]
I agree, this would be very valuable. It tripped up everyone I know working with these examples.

**Comment by sguada** [comment]
Have you take a look at the [notebook](https://github.com/tensorflow/models/blob/master/slim/slim_walkthough.ipynb)

**Comment by abes975** [comment]
Thanks a lot, but I did still not understand whether the slim.train can handle automatically mini batch of data or there is the need for and explicit loop. It seems that in the example (the one of the convolutional neural network) only one minibatch of data is used for training and only for one iteration. Is it possible to clarify that?
Unfortunately the data set used is assumed to be already in TFrecord format.
The following code (Taken from the notebook) so all function are specified
Do train operation on **entire** dataset or **just on a single batch** so an explicit loop has to be added?


`with tf.Graph().as_default():
    dataset = flowers.get_split('train', flowers_data_dir)
    images, _, labels = load_batch(dataset)
    # Create the model:
    logits = my_cnn(images, num_classes=dataset.num_classes, is_training=True)
  
<SNIP> all operation to hot one encode the lables create optimizer..taken from the notebook in the link 

    # Run the training:
    final_loss = slim.learning.train(
      train_op,
      logdir=train_dir,
      number_of_steps=1, # For speed, we just do 1 epoch
      save_summaries_secs=1)`

**Comment by sguada** [comment]
slim.train can run forever, just leave number_of_steps=None, or specify how many train steps you want to run. So for every step it would train in one batch.

**Comment by abes975** [comment]
These answer were very helpful!
Thanks a lot!

**Comment by AKSHAYUBHAT** [comment]
In spite of knowing python and tensorflow well, I still have difficult time understanding the slim library / models. It started out as an effort to simplify TensorFlow, but now its even more difficult to understand than tensorflow itself.

1. Why is the slim split across three different places (tf.contrib.slim,models/slim and (deprecated) models/inception/slim ) ? Can you guys consider renaming models/slim to models/slim_models to avoid confusion?

2. The now deprecated models/slim had a beautiful simple functional interface, and but why was that approach abandoned for slim.dataset.?? 

3. I understand the code in this particular directory is written for enabling `eval_image_classifier.py` etc. to work, but its quite difficult to understand. E.g. why create different classes for each dataset. or have the structure folder name -> label mapping. It might be simple to use a text or a json file, with list of all paths and another one with labels, rather than relying on some directory path pattern matching.

4. Is it possible to provide a simple example of loading batch data?  I saw that there is load_batch in the notebook, but it uses flower dataset, which involves "provider" pattern with `slim.dataset_data_provider.DatasetDataProvider(dataset,....` which gets even more difficult to parse since "dataset" class involves a "file pattern" rather than a simple list of files. 

**I appreciate the hard work put into this Open Source code, but honestly even after having several years worth of ML and Python experience. The code in this repo (models/slim) is frankly indecipherable, and reads like it was written for Java, while the underlying slim library being designed to provide a simple functional interface (at least for nets). Extending code written in this repo to work with any other dataset is actually more complex than starting from scratch, and while the initial goal would have been to provide a set of scripts such as 'eval_image_classifier.py' to work with any model / dataset its actually more confusing.**

**Comment by AKSHAYUBHAT** [comment]
Also why is the nets code duplicated. E.g. its their in tf.contrib.slim.python.slim.nets  as well as in models/slim/nets ? Which one is actually being used?

**Comment by ravijagan** [comment]
I dont have much experience like akshayubhat. Someone suggested using this tfslim and it is now a nightmare after 3 months. tensorflow may have been longer route, but it was documented mapped journey. This is a shortcut through a dense forest full of circular references.

**Comment by yifeif** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
